featuretools/entityset/entityset.py

```diff
@@ -159,67 +158,65 @@ def __getitem__(self, dataframe_name):
     def dataframes(self):
         return list(self.dataframe_dict.values())
 
-# --> Add back later: needs to wait till serialization is implemented
-    # @property
-    # def metadata(self):
-    #     '''Returns the metadata for this EntitySet. The metadata will be recomputed if it does not exist.'''
-    #     if self._data_description is None:
-    #         description = serialize.entityset_to_description(self)
-    #         self._data_description = deserialize.description_to_entityset(description)
+    @property
+    def metadata(self):
+        '''Returns the metadata for this EntitySet. The metadata will be recomputed if it does not exist.'''
+        if self._data_description is None:
+            description = serialize.entityset_to_description(self)
+            self._data_description = deserialize.description_to_entityset(description)
 
-    #     return self._data_description
+        return self._data_description
```

Comment:
If metadata an entityset where the DataFrames are just a TableSchema on an empty DataFrame, I feel like it should be possible to create that without having to go through serialization and have this `empty_dataframe` function.

Here's one idea:
```python
    @property
    def metadata(self):
        '''Returns the metadata for this EntitySet. The metadata will be recomputed if it does not exist.'''
        if self._data_description is None:
            metadata_es = EntitySet(self.id)
            for df in self.dataframes:
                empty_df = df.loc[[], :]
                empty_df.ww.init(schema=df.ww.schema, validate=False)
                metadata_es.add_dataframe(empty_df)

            metadata_es.add_relationships(self.relationships)
            self._data_description = metadata_es

        return self._data_description
```

Reply:
While I think this is a decent idea, it has a couple complications that might drive it beyond the scope of this PR:
- The current deserialization to empty dataframes always uses pandas dataframes. This approach would also create Dask and Koalas empty dataframes - based on some limited testing this was causing additional errors that would need to be fixed
- At least a couple tests rely on deserialization into an empty dataframe - so we would either need to keep that capability in deserialize or refactor those tests.

@tamargrey  I think it would be best if we set this aside for now and create an issue to revisit later if you are ok with that?

Reply:
Ah yeah if it's going to create a bunch of additional things that need to be handled, definitely not worth going into now. Happy to set aside for now

Reply:
Oh, one more thing - as I was looking over the `empty_dataframe` function, I did see some code there that I believe was not necessary as it wasn't being used, so I removed that.

As I was doing that I also realized that the metadata entityset won't have the categories stored in the column dtypes like the main entityset does. I'm not sure if this will create problems elsewhere or not?

Reply:
ah... interesting. I guess I'm not entirely clear on what the metadata gets used for, but if we did need the categories, we could possibly store them elsewhere in the column schema or something.

Reply:
Yeah, we have the information available in the json to recreated, but we would need to implement some additional code to to cast the columns properly in `empty_dataframe`. When deserializing with data, the dtypes are set during the read from disk calls, but since we aren't doing that in `empty_dataframe` the dtypes get set during Woodwork initialization.

As long as we don't need to access the category values in the metadata I think we should be fine, but I'm not entirely sure where all it gets used either...

Reply:
Created issue #1476 for looking into removing `empty_dataframe`

Reply:
After thinking about this a bit more, there was an easy way to recreate the category dtypes including categories for the empty dataframe, so I added this code.